### PR TITLE
AP_RangeFinder: Add JRE-30 radar support

### PR DESCRIPTION
**Pull request by rmackay9**
This adds support for the JAE-30 radar which is a 30m radar from Japan Aviation Electronics.

This replaces PR https://github.com/ArduPilot/ardupilot/pull/25377 but I've made a few mostly small NFC tweaks including:

1. shortened the DIST_MAX_CM definition from 50000cm to 5000cm (e.g. 50m).  This is the max distance shown when out-of-range.
2. removed an unused variable
3. added some comments
4. changed the crc loop to use uint32_t instead of "int"
5. only enabled on boards with > 1MB of flash

I've tested this fairly thoroughly including confirming the status changes correctly including when the radar is pointed up into the sky where it sees nothing.

BTW, during the review of the previous PR we decided to re-work the state machine.  The original author went ahead and did that and although it's still a little different from how we normally write these parsers I think it's close enough.

The corresponding wiki PR is here https://github.com/ArduPilot/ardupilot_wiki/pull/5546

### CODE REVIEW

**Comment by rishabsingh3003** [comment]
We usually have included a simulated rangefinder for all the recent serial rangefinders that were merged

**Comment by rmackay9** [comment]
PR of JRE simulator: https://github.com/ArduPilot/ardupilot/pull/25532

File: libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        // ensure we have a packet type:
+        if (data_buff_idx < 2) {
+            continue;
+        }
+
+        // determine packet length for incoming packet:
+        uint8_t packet_length;
+        switch (data_buff[1]) {
+        case 'A':
```
Comment (rmackay9) [inline libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp:87]: @peterbarker should we re-use the header definitions?  e.g. FRAME_HEADER_2_A.

Comment (peterbarker) [reply libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp:87]: Just remove the header definitions IMO.

Unless you use the define in more than one place the only thing they add is an unnecessary level of indirection (i.e. to know what value you need to find in your diagnostic output you have to leave the code you're currently stepping through to find the `#define`.

Comment (rmackay9) [reply libraries/AP_RangeFinder/AP_RangeFinder_JRE_Serial.cpp:87]: I don't feel too strongly about it but higher up in the move_preamble_in_buffer() function you've used the definition so I think I'll change these lines to also use the definition.  I think the definition names are slightly better than the magic characters simply appearing in the code although the same could probably be achieved by adding a comment.

**Comment by rmackay9** [comment]
I've re-tested and it all seems to work fine, thanks @peterbarker!

**Review by tridge — APPROVED** [review]

**Pull request merged**